tests/manage/z_cluster/cluster_expansion/conftest.py

```diff
@@ -107,3 +110,40 @@ def factory(ocs_nodes=False, node_count=3, taint_label=None):
         log.info(f"Successfully Tainted nodes {new_nodes} with {taint_label}")
 
     return factory
+
+
+@pytest.fixture
+def add_capacity_setup(add_nodes):
+    """
+    Check that we have the right configurations before we start the test
+    """
+    log.info("Start add capacity setup")
+    osd_pods_before = get_osd_pods()
+    number_of_osd_pods_before = len(osd_pods_before)
+    if number_of_osd_pods_before >= constants.MAX_OSDS:
+        pytest.skip("We have maximum of OSDs in the cluster")
+
+    # If we use vSphere we may need to add more worker nodes
+    # to the cluster before starting the test
+    if (
+        config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM
+        and number_of_osd_pods_before >= 9
+    ):
+        num_of_expected_wnodes = 6
```

Comment:
i feel this should not be hardcoded we can what if add capacity is done more then 1 time

Reply:
I don't think we need to run the "add capacity" test with more than 6 worker nodes.